js/indicia.datacomponents/jquery.idc.filterSummary.js

```diff
@@ -193,6 +193,47 @@
         });
       }
 
+      // es-filter-param controls
+      if ($('.es-filter-param').length) {
+        $.each($('.es-filter-param'), function eachParam() {
+          var val = $(this).val();
+          // Skip if no value.
+          if (val === null || val.trim() === '') {
+            return;
+          }
+          // Skip if unchecked checkbox
+          if ($(this).is(':checkbox') && !$(this).is(':checked')) {
+            return;
+          }
+   
+          var esBoolClause = indiciaFns.getDataValueFromInput(this, 'data-es-bool-clause');
+          var esField = indiciaFns.getDataValueFromInput(this, 'data-es-field');
+          var esQueryType = indiciaFns.getDataValueFromInput(this, 'data-es-query-type');
+          var esQuery = indiciaFns.getDataValueFromInput(this, 'data-es-query');
+          var esNested = indiciaFns.getDataValueFromInput(this, 'data-es-nested');
+
+          html = addHtml(html, '<div><b>ES query: </b>');
+          if (esField) {
+             html = addHtml(html, esField + ' ');
+          }
+          if (esBoolClause) {
+             html = addHtml(html, esBoolClause + ' ');
+          }
+          if (esQueryType) {
+             html = addHtml(html, esQueryType + ' ');
+          }
+          if (esNested) {
+             html = addHtml(html, esNested + ' ');
+          }
+          if (esQuery) {
+             html = addHtml(html, esQuery.replace('#value#', '<strong>' + val + '</strong>'));
+          } else {
+             html = addHtml(html, '<strong>' + val + '</strong>');  
+          }
+          html = addHtml(html, '</div>');
```

Comment:
I'm guessing that the filter string displayed by this will need a bit of experience to understand. OK for now I think but we should review whether it needs to be made more like readable English.

Reply:
Hi John. I knew this wasn't ideal, but because of the flexibility specifying an ES query, I envisaged some very tricky parsing. However, your comment made me think again and I've come up with a better approach which is to allow a new parameter - 'data-es-summary' where form creator can put in a meaningful summary string with a #value# token which is replaced with the specified value - much like the data-es-query. If data-es-summary is not specified, then it resorts to the method I coded originally.

Reply:
Thanks @burkmarr - please make sure that data-es-summary ends up in the documentation on read the docs at some point.